Marlin/Marlin_main.cpp

```diff
@@ -596,6 +596,43 @@ void setup_powerhold() {
   #endif
 }
 
+/*
+  Enable 24V to fans, E-reg, Cartridge Holder, Cartridges
+  if no short circuit is detected on line
+  ~200us, blocking time
+  Requires serial to be enabled to generate error condition
+*/
+void enable_power_supply() {
+  // Turn on FAN first so that 24V does not have a path to 24V_SW through the Rambo. TODO: Disconnect these two circuits.
+  pinMode(FAN_CHASSIS_TOP_PIN, OUTPUT);
+  digitalWrite(FAN_CHASSIS_TOP_PIN, HIGH);
+  delay(6);  // Delay for line to settle. 
+  
+  analogRead(PS_MONITOR_PIN);  // Throw out first reading
+  int V_Monitor_Result = analogRead(PS_MONITOR_PIN);
+  if (V_Monitor_Result > PS_ENABLE_LOWER_LIMIT)   // Test for 24V short to gnd
+  {
+    if (V_Monitor_Result <= PS_ENABLE_UPPER_LIMIT) // Test for 24V short to 5V
+    {
+      digitalWrite(PS_FORCE_ON_LL,LOW); // Pull Power Supply enable low to force on
+      pinMode(PS_FORCE_ON_LL, OUTPUT);  //
+      delayMicroseconds(100);           // 100us minimum delay to register on
+      pinMode(PS_FORCE_ON_LL, INPUT);   // Release Power Supply enable to allow current limiting
```

Comment:
There is an external pull-up that sets the default "high" state. All we have to do is pull it down and let go.

There is a delay between pin-modes to ensure the 24V_SW FET has time to turn on. If we go too fast, the switch will never fully enhance (because of the gate capacitance), and the circuit will just collapse back to off. Time constant is roughly 10000pF*10k = 100us. I'll increase the delay to 200us and add this to the comments. It is also a balance, because while we have this line low, there is no current protection. Holding low for a significant portion of time will increase our chances of smoking something if the detection circuit failes. Lets probe this on Monday.